svi/auto_thermal_reformer/run_fullspace_sweep.py

```diff
@@ -70,47 +79,65 @@ def main(X,P):
     else:
         # If the solver didn't converge, we don't care about the solve time,
         # the objective, or any of the degree of freedom values.
-        for key in config.PARAM_SWEEP_KEYS:
-            if key not in ("X", "P", "Termination"):
+        for key in df_keys:
+            if key not in ("X", "P", "Termination", "Condition Number"):
                 df[key].append(INVALID)
 
 
 if __name__ == "__main__":
+
     argparser = config.get_sweep_argparser()
+
     argparser.add_argument(
         "--fname",
         default="fullspace-sweep.csv",
         help="Base file name for parameter sweep results"
     )
+
+    argparser.add_argument(
+        "--calc-condition-number",
+        action="store_true",
+        help="Whether to calculate the condition number or not"
+    )    
+
     args = argparser.parse_args()
+
+    df = {key: [] for key in config.PARAM_SWEEP_KEYS}
+    index_to_insert = 3
+    new_key = "Condition Number" if args.calc_condition_number else None
+    df_keys = list(df.keys())
```

Comment:
Can you change `df_keys` to `DF_KEYS` to indicate that it is a global variable (with respect to the `main` function).